scipy/stats/_continuous_distns.py

```diff
@@ -832,8 +832,27 @@ def fit(self, data, *args, **kwds):
         return a, b, floc, fscale
 
     def _entropy(self, a, b):
-        return (sc.betaln(a, b) - (a - 1) * sc.psi(a) -
-                (b - 1) * sc.psi(b) + (a + b - 2) * sc.psi(a + b))
+        def regular(a, b):
+            return (sc.betaln(a, b) - (a - 1) * sc.psi(a) -
+                    (b - 1) * sc.psi(b) + (a + b - 2) * sc.psi(a + b))
+
+        def asymptotic_ab_large(a, b):
+            sum_ab = a + b
+            log_term = 0.5 * (
+                np.log(2 * np.pi * a * b) - 3 * np.log(sum_ab) + 1
+            )
+            t1 = (
+                (110 * sum_ab**3 + 20 * sum_ab**2 + sum_ab - 2)
+                / (120 * sum_ab**4)
+            )
+            t2 = (1 - 50 * a**3 - 10 * a**2 - a) / (120 * a**4)
```

Comment:
The main comment is to avoid raising to positive powers here because they can overflow, and then you end up with `inf/inf`.